### PR TITLE
Publish core using publish.js and improve packaging

### LINKED ISSUE #1973: [Housekeeping] Store core distribution bundle with designsystem
- [X] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the housekeeping request

Distribution bundles for "core" and "designsystem" libs when running `npm run build` are outputted to respectively: 
- `designsystem/libs/core/dist` 
- `designsystem/dist/libs/designsystem` 

To keep it consistent and seperate source code from distribution bundles i would like for the core distribution bundle to be outputted to `designsystem/dist/libs/core` instead. 

`scripts/forward-scss-files` will probably have to be updated as part of this.

### LINKED ISSUE #1974: [Housekeeping] Copy SCSS files from core on publish instead of dev time
- [X] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the housekeeping request
Currently the gitignored folder `libs/core/scss` which contains the distributed sass files for core is generated whenever stencil builds. This includes when running the development server. 

This is unnecessary as the files are never accessed during development and should be used for distribution only. Honestly it is also annoying as it is easy to mistake `libs/core/scss` for `libs/core/src/scss` when developing causing confusion. 

The code responsible for copying out the sass files can be found in `libs/core/stencil.config.ts`: 
```
    {
      type: 'dist',
      esmLoaderPath: '../loader',
      copy: [{ src: 'scss', dest: '../../scss', warn: true }],
    },
```

The copy key should be removed and it should be done as part of `scripts/publish.js` instead. 

## Are there any additional context?

The location of `libs/core/scss` might have changed with #1973

### PR DESCRIPTION
**Pull request by mictro**
## Which issue does this PR close?

This PR closes #1973 #1974 #1975

## What is the new behavior?

Publish core package with the publish script publish.js using the dist/libs/core distribution folder. The core SCSS-files only gets copied to the distribution on publish. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [X] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [X] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
There is a bit more needed before this will be working as intended. 

As the [failing build log shows](https://github.com/kirbydesign/designsystem/runs/7516274404?check_suite_focus=true#step:3:116), a scss folder is expected to be present within the core library folder, as this is a way to let npm know that this should be copied into in the final published package.

As the folder no longer exists on this location, it will now not be automatically included by npm when we run the core publish action, and therefore we probably need to handle the publishing in this script as well, similar to how it is done for the designsystem.

File: scripts/publish.js [hunk 1]
```diff
@@ -183,14 +208,119 @@ function publish() {
   }
 }
 
+function cleanCoreDistribution() {
```
Comment (RasmusKjeldgaard) [inline scripts/publish.js:211]: Could we consider parameterisation to combine the more or less identical functions into one?

Comment (RasmusKjeldgaard) [reply scripts/publish.js:211]: I guess this goes for most of the newly added stuff.

Comment (mictro) [reply scripts/publish.js]: Yes, indeed - done!

File: scripts/publish.js [hunk 2]
```diff
@@ -84,113 +85,168 @@ function cleanDistribution() {
...
-
+function createScssCoreForwardFiles(coreLibSrcDir, sccsCoreForwardFilePaths) {
   const sourceRootDir = `${coreLibDir}/scss`;
   const sharedRootDir = libsRootDir;
 
-  return new Promise((resolve) => {
-    pathsToForwardCoreScssFilesTo.forEach((targetRootDir) => {
-      forwardScssFiles({ sourceRootDir, targetRootDir, packageAlias, sharedRootDir });
-      resolve();
+  const copySccsCoreTargetFiles = () => {
```
Comment (RasmusKjeldgaard) [inline scripts/publish.js:128]:
```suggestion
  const copyScssCoreTargetFiles = () => {
```

File: scripts/publish.js [hunk 3]
```diff
@@ -84,113 +85,168 @@ function cleanDistribution() {
...
+    console.log('Creating SCSS core forwarding files...');
+    return new Promise((resolve) => {
+      sccsCoreForwardFilePaths.forEach((targetRootDir) => {
+        forwardScssFiles({ sourceRootDir, targetRootDir, packageAlias, sharedRootDir });
+        resolve();
+      });
+    });
+  };
+
+  const deleteSccsCoreTargetFiles = () => {
```
Comment (RasmusKjeldgaard) [inline scripts/publish.js:147]:
```suggestion
  const deleteScssCoreTargetFiles = () => {
```

File: scripts/publish.js [hunk 4]
```diff
@@ -84,113 +85,168 @@ function cleanDistribution() {
...
+    console.log('Deleting SCSS core target files...');
+    return fs.remove(`${sourceRootDir}`);
+  };
+
+  return copySccsCoreTargetFiles()
```
Comment (RasmusKjeldgaard) [inline scripts/publish.js:152]:
```suggestion
  return copyScssCoreTargetFiles()
```

File: scripts/publish.js [hunk 5]
```diff
@@ -84,113 +85,168 @@ function cleanDistribution() {
...
+    .then(generateSccsCoreForwardFiles)
+    .then(deleteSccsCoreTargetFiles);
```
Comment (RasmusKjeldgaard) [inline scripts/publish.js:154]:
```suggestion
    .then(generateScssCoreForwardFiles)
    .then(deleteScssCoreTargetFiles);
```

File: scripts/publish.js [hunk 6]
```diff
@@ -84,113 +85,168 @@ function cleanDistribution() {
...
     path.extname(input) === '' || input.endsWith('-loader.js') || input.endsWith('.min.js');
-  return fs.copy(`${angularLibSrcDir}/polyfills`, `${distTarget}/polyfills`, {
+  return fs.copy(`${libSrcDir}/polyfills`, `${distTarget}/polyfills`, {
     filter: onlyLoadersAndMinified,
   });
 }
 
-function createTarballPackage() {
+function copyCoreStencilBuildFiles(stencilBuildDir, distTarget) {
+  console.log('Copying core stencil build files...');
```
Comment (RasmusKjeldgaard) [inline scripts/publish.js:175]: Consider renaming this, as there is no stencil-specifics at this time. 
It should all be compiled to something the browser understands, so maybe copyCoreDistributionFiles is a better name?

```suggestion
function copyCoreDistributionFiles(stencilBuildDir, distTarget) {
  console.log('Copying core distribution files...');
```

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
Looks awesome.

Just added some suggestions and fixes for typos.

**Review by RasmusKjeldgaard — DISMISSED** [review]
🚀 

If we sync with develop, I think we are good to go as publish.yml should be deleted 👍

**Review by RasmusKjeldgaard — APPROVED** [review]

**Pull request merged**